modules/HelpDesk/models/Record.php

```diff
@@ -74,4 +74,58 @@ public function getHierarchyDetails(): array
 		}
 		return $hierarchy;
 	}
+
+	/**
+	 * Function check if record can be closed.
+	 *
+	 * @param string $status
+	 *
+	 * @return array
+	 */
+	public function checkValidateToClose(string $status): array
+	{
+		if (\in_array($status, \App\RecordStatus::getStates($this->getModuleName(), \App\RecordStatus::RECORD_STATE_CLOSED))) {
```

Comment:
check first
\App\Config::module($this->getModuleName(), 'CHECK_IF_RECORDS_HAS_TIME_CONTROL')
\App\Config::module($this->getModuleName(), 'CHECK_IF_RELATED_TICKETS_ARE_CLOSED')